File: files/problems/cpp/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int lastB=-1,lastW=-1;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(lastB==-1 && nums[i]==2){
                lastB=i;
            }
            else if(lastB!=-1 && nums[i]!=2){
                    swap(nums[lastB],nums[i]);
                    lastB += 1;
            }
        }
        for(int i=0;i<n;i++){
            if(i==lastB){
                break;
            }
            if(lastW==-1 && nums[i]==1){
                lastW=i;
            }
            else if(lastW!=-1 && nums[i]!=1){
                    swap(nums[lastW],nums[i]);
                    lastW += 1;
            }
        }
        return;
    }
};

File: leetcode/problems/java/sort-colors.java
class Solution {
  public void sortColors(int[] nums) {
    int zero = 0, one = 0;
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] == 0) {
        zero++;
      } else if (nums[i] == 1) {
        one++;
      }
    }
    for (int i = 0; i < nums.length; i++) {
      if (zero > 0) {
        nums[i] = 0;
        zero--;
      } else if (one > 0) {
        nums[i] = 1;
        one--;
      } else {
        nums[i] = 2;
      }
    }
  }
}